File: python/task1_1.py
# coding=UTF-8
# 1.1. Conversão RGB-YIQ-RGB
# (cuidado com os limites de R, G e B!)

import numpy as np
import utils


def main():
    image = utils.load_image('lenna.png')

    yiq = utils.rgb2yiq(image)
    back_to_original = utils.yiq2rgb(yiq)

    utils.display_single_image('Original', image)
    utils.display_single_image('YIQ', yiq)
    utils.display_single_image('Back to Original', back_to_original)

    utils.save_image('yiq.png', yiq)

    utils.wait_key_and_destroy_windows()


if __name__ == "__main__":
    main()

File: matlab/task1_1.m
% 1.1 Conversão RGB-YIQ-RGB
% (cuidado com os limites de R, G e B!)
function task1_1()
    assetsDir = ['..' filesep 'assets' filesep];
    filePath = strcat(assetsDir, 'lenna.png');
    
    originalImage = imread(filePath);    
    yiqImage = rgb2ntsc(originalImage);
    back2RGB = ntsc2rgb(yiqImage);
    
    % Ploting
    fig=figure();    
    subplot(1,3,1), imshow(originalImage,[]), title('Original image')
    subplot(1,3,2), imshow(yiqImage,[]), title('YIQ image')
    subplot(1,3,3), imshow(back2RGB,[]), title('Back to RGB image')
    strcat('output',filesep,mfilename,'.png')
    
    saveas(fig,strcat('output',filesep,mfilename,'.png'));
end